build-tools/packages/build-tools/src/fluidBuild/fluidPackageCheck.ts

```diff
@@ -492,34 +492,77 @@ export class FluidPackageCheck {
             // const hasPrettier = pkg.getScript("prettier");
             // const lintChildren = hasLint ? ["prettier", "eslint"] : ["eslint"];
 
-            const lintScript = pkg.getScript("lint");
-            const lintFixScript = pkg.getScript("lint:fix");
+            /* Policy for "lint" & "lint:fix" scripts
+                1. Must end with "npm run eslint" / "npm run eslint:fix" respectively
+                2. Must contain "npm run prettier" / "npm run prettier:fix" respectively
+            */
+            let lintScript = pkg.getScript("lint");
+            let lintFixScript = pkg.getScript("lint:fix");
+
+            const prettier = "npm run prettier";
+            const prettierFix = "npm run prettier:fix";
+
+            const hasPrettier = lintScript?.includes(prettier);
+            const hasPrettierFix = lintFixScript?.includes(prettierFix);
 
             const expectedEndScript = "npm run eslint";
             const expectedEndScriptFix = "npm run eslint:fix";
 
-            if (lintScript === undefined || !lintScript.endsWith(expectedEndScript)) {
+            const endsWithEslint = lintScript?.endsWith(expectedEndScript);
+            const endsWithEslintFix = lintFixScript?.endsWith(expectedEndScriptFix);
+
+            // check lint script
+            if (!(hasPrettier && endsWithEslint)) {
```

Comment:
Since this logic has gotten a bit more complex, and it is more-or-less duplicated for "lint" and "lint:fix", it might be nice to extract the logic into a small helper function (would recommend making that function private to this method, since the logic is very specific to this method).

Reply:
I created a separate helper function `checkLintScriptsHelper`. `checkLintScripts` passes in the boolean value `isLint` to differentiate between `lint` and `lint:fix`. 

Would like your opinion if this would be readable for other engineers in the future.

Reply:
Ignore the comment above, made changes by passing in the string variable instead of boolean value.